sso-url-helper/src/main/java/it/niedermann/nextcloud/sso/urlhelper/UrlHelper.kt

```diff
@@ -0,0 +1,132 @@
+@file:JvmName("UrlHelper")
+
+package it.niedermann.nextcloud.sso.urlhelper
+
+import android.content.Context
+import android.text.TextUtils
+import android.util.Log
+import com.nextcloud.android.sso.QueryParam
+import com.nextcloud.android.sso.helper.SingleAccountHelper
+import com.nextcloud.android.sso.model.SingleSignOnAccount
+import java.net.MalformedURLException
+import java.net.URL
+import java.util.LinkedList
+import java.util.Optional
+
+class FallbackPreviewSize(val width: Int, val height: Int)
+val Context.fallbackPreviewSize get() = FallbackPreviewSize(
+    width = resources.displayMetrics.widthPixels,
+    height = resources.displayMetrics.heightPixels,
+)
+
+/**
+ * Generates an [URL] object from the given [model]
+ * @throws [IllegalArgumentException] in case the [model] can not be processed or the [ssoAccount] is not valid for the [model]
+ */
+fun getAbsoluteUrl(ssoAccount: SingleSignOnAccount, model: String, fallbackPreviewSize: FallbackPreviewSize): URL {
+    return try {
+        // Check whether this is a complete URL, will throw an MalformedURLException if not
+        val url = URL(model)
+        // Verify it starts with the given ssoAccount
+        if (model.startsWith(ssoAccount.url)) {
+            rewriteSpecialURLs(ssoAccount, url, fallbackPreviewSize).orElse(url)
+        } else {
+            throw IllegalArgumentException("Given ${SingleSignOnAccount::class.java.simpleName} does not match the URL (${ssoAccount.url} vs. ${model}). Pass correct ${SingleSignOnAccount::class.java.simpleName} or use correct URL (or a plain ${String::class.java.simpleName}) to try fetching with the current ${SingleSignOnAccount::class.java.simpleName} stored in ${SingleAccountHelper::class.java.simpleName}.")
```

Comment:
`or use correct URL` -> I replaced the glide reference here but was not really sure what to write here.